crates/biome_js_analyze/src/lint/nursery/no_react_specific_props.rs

```diff
@@ -0,0 +1,101 @@
+use biome_analyze::context::RuleContext;
+use biome_analyze::{declare_rule, ActionCategory, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make::{jsx_ident, jsx_name};
+use biome_js_syntax::{AnyJsxAttributeName, JsxAttribute};
+use biome_rowan::{AstNode, BatchMutationExt, TextRange};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Prevents React-specific JSX properties from being used.
+    ///
+    /// This rule is intended for use in JSX-based frameworks (mainly Solid.js)
+    /// that do not use React-style prop names.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// <Hello className="John" />
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// <Hello class="Doe" />
+    /// ```
+ pub NoReactSpecificProps {
+        version: "1.0.0",
+        name: "noReactSpecificProps",
+        sources: &[RuleSource::EslintSolid("no-react-specific-props")],
+        recommended: false,
+    }
+}
+
+const REACT_SPECIFIC_JSX_PROPS: &[&str] = &["className", "htmlFor"];
+
+fn get_replacement_for_react_prop(str: &str) -> &'static str {
+    match str {
+        "className" => "class",
+        "htmlFor" => "for",
+        _ => panic!("should be a React prop: {str:?}"),
+    }
+}
+
+impl Rule for NoReactSpecificProps {
+    type Query = Ast<JsxAttribute>;
+    type State = (TextRange, &'static str);
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let attribute = ctx.query();
+        let name = attribute.name().ok()?;
+        let range = name.range();
+        let name = name.text();
+
+        if REACT_SPECIFIC_JSX_PROPS.contains(&name.as_str()) {
+            Some((range, get_replacement_for_react_prop(&name)))
+        } else {
+            None
+        }
+    }
+
+    fn diagnostic(
+        _: &RuleContext<Self>,
+        (range, replacement): &Self::State,
+    ) -> Option<RuleDiagnostic> {
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                range,
+                markup!("This JSX property is specific to React."),
+            )
+            .detail(
+                range,
+                format!("Replace this attribute name with {replacement:?}"),
```

Comment:
This second message should explain the error. It should explain why React-specific props are an error.

The `Replace this attribute name with{replacement:?}` should be moved inside the `action` function